woocommerce/woocommerce
# Product Gallery: reset main image when variations are cleared

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/42792

This PR resets the main image when variations are cleared.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Appearance > Editor > Templates > Single Product
2. Clear customizations if needed and then convert to blocks
3. Add Product Gallery block and save
4. Go to a product with variations such as Hoodie in the frontend
5. Select variation Green and No (notice main image changes to green shirt)
6. Now click on "Clear"
7. Ensure the main image changes back to the default original main image (red shirt)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
